util/util.go

```diff
@@ -0,0 +1,101 @@
+package util
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/prometheus/common/promlog"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+)
+
+var (
+	// Application wide logger
+	logger log.Logger
+)
+
+func InitLogger(logLevel string) {
+	allowedLevel := promlog.AllowedLevel{}
+	allowedLevel.Set(logLevel)
+	logger = promlog.New(allowedLevel)
+}
+
+func LogInfo(keyvals ...interface{}) {
+	level.Info(logger).Log(keyvals...)
+}
+
+func LogWarn(keyvals ...interface{}) {
+	level.Warn(logger).Log(keyvals...)
+}
+
+func LogError(keyvals ...interface{}) {
+	level.Error(logger).Log(keyvals...)
+}
+
+func LogDebug(keyvals ...interface{}) {
+	level.Debug(logger).Log(keyvals...)
+}
+
+//ThroughtputCalc runs on scheduled interval to calculate the throughput per second and sends results to a channel
+type ThroughtputCalc struct {
+	tickInterval time.Duration
+	previous     float64
+	current      chan float64
+	Values       chan float64
+	running      bool
+	lock         sync.Mutex
+}
+
+func NewThroughputCalc(interval time.Duration) *ThroughtputCalc {
+	return &ThroughtputCalc{tickInterval: interval, current: make(chan float64, 1), Values: make(chan float64, 1)}
+}
+
+func (dt *ThroughtputCalc) SetCurrent(value float64) {
+	select {
+	case dt.current <- value:
+	default:
+	}
+}
+
+func (dt *ThroughtputCalc) Start() {
+	dt.lock.Lock()
+	defer dt.lock.Unlock()
+	if !dt.running {
+		dt.running = true
+		ticker := time.NewTicker(dt.tickInterval)
+		go func() {
+			for range ticker.C {
+				if !dt.running {
+					return
+				}
+				current := <-dt.current
+				diff := current - dt.previous
+				dt.previous = current
+				select {
+				case dt.Values <- diff / dt.tickInterval.Seconds():
+				default:
+				}
+			}
+		}()
+	}
+}
+
+// Blocking retry with a fixed delay
+func RetryWithFixedBackoff(retries int, wait time.Duration, f func() (interface{}, error)) (interface{}, error) {
```

Comment:
Make `retries` an uint. Not sure it makes sense to be able to pass negative values.